system-contracts/test/L1Messenger.spec.ts

```diff
@@ -0,0 +1,517 @@
+      )
+        .to.emit(l1Messenger, "L2ToL1LogSent")
+        .withArgs([
+          0,
+          logData.isService,
+          1,
+          l1MessengerAccount.address,
+          ethers.utils.hexlify(logData.key),
+          ethers.utils.hexlify(logData.value),
+        ]);
+    });
+
+    it("should emit L2ToL1LogSent event when called by the system contract with isService false", async () => {
+      await expect(
+        l1Messenger
+          .connect(l1MessengerAccount)
+          .sendL2ToL1Log(false, ethers.utils.hexlify(logData.key), ethers.utils.hexlify(logData.value))
+      )
+        .to.emit(l1Messenger, "L2ToL1LogSent")
+        .withArgs([
+          0,
+          false,
+          1,
+          l1MessengerAccount.address,
+          ethers.utils.hexlify(logData.key),
+          ethers.utils.hexlify(logData.value),
+        ]);
+    });
+
+    it("should revert when called by the system contract with empty key & value", async () => {
+      const emptyKV = ethers.utils.hexlify([]);
+      await expect(l1Messenger.connect(l1MessengerAccount).sendL2ToL1Log(logData.isService, emptyKV, emptyKV)).to.be
+        .rejected;
+    });
+
+    it("should revert when called by the system contract with key & value > 32 bytes", async () => {
+      const oversizedKV = ethers.utils.hexlify(randomBytes(33));
+      await expect(l1Messenger.connect(l1MessengerAccount).sendL2ToL1Log(logData.isService, oversizedKV, oversizedKV))
+        .to.be.rejected;
+    });
+
+    it("should revert when called by the system contract with key & value < 32 bytes", async () => {
+      const undersizedKV = ethers.utils.hexlify(randomBytes(31));
+      await expect(l1Messenger.connect(l1MessengerAccount).sendL2ToL1Log(logData.isService, undersizedKV, undersizedKV))
+        .to.be.rejected;
+    });
+  });
+
+  describe("sendToL1", async () => {
+    it("should emit L1MessageSent & L2ToL1LogSent events", async () => {
+      const expectedKey = ethers.utils
+        .hexZeroPad(ethers.utils.hexStripZeros(l1MessengerAccount.address), 32)
+        .toLowerCase();
+      await expect(l1Messenger.connect(l1MessengerAccount).sendToL1(logData.message))
+        .to.emit(l1Messenger, "L1MessageSent")
+        .withArgs(l1MessengerAccount.address, ethers.utils.keccak256(logData.message), logData.message)
+        .and.to.emit(l1Messenger, "L2ToL1LogSent")
+        .withArgs([0, true, 1, l1Messenger.address, expectedKey, ethers.utils.keccak256(logData.message)]);
+    });
+  });
+
+  describe("requestBytecodeL1Publication", async () => {
+    it("should revert when not called by known code storage contract", async () => {
+      const byteCodeHash = ethers.utils.hexlify(randomBytes(32));
+      await expect(l1Messenger.requestBytecodeL1Publication(byteCodeHash)).to.be.rejectedWith("Inappropriate caller");
+    });
+
+    it("shoud revert when byteCodeHash < 32 bytes, called by known code system contract", async () => {
```

Comment:
The same as about similar to tests for `sendL2ToL1Log`. I am not sure what is actually tested here

Reply:
fixed